1. Developer basics/6_ides/6-1_intellij_vscode_shortcuts.md

```diff
@@ -0,0 +1,146 @@
+| Open Settings                                                                                          | `Ctrl + Alt + S`                |
+| Go to next/previous editor tab                                                                         | `Alt + Right / Alt + Left`      |
+| Close Active tab/window                                                                                | `Ctrl + Shift + F4 / Ctrl + F4` |
+
+### Find Everything
+
+| Functions                      | Shortcut                      |
+| ------------------------------ | ----------------------------- |
+| Search everywhere              | `Shift Shift`                 |
+| Find/replace                   | `Ctrl + F / R`                |
+| Find in path/Replace in path   | `Ctrl + Shift + F / R`        |
+| Next/previous occurence        | `F3 / Shift + F3`             |
+| Find word at caret             | `Ctrl + F3`                   |
+| Go to class/file               | `Ctrl + N / Ctrl + Shift + N` |
+| Go to function                 | `Ctrl + Alt + Shift + N`      |
+
+All these **Shortcuts** are the most commonly used, however, Jetbrains IDEs allows use a huge list of shortcuts to manage all the things with your keyboard, here is a [Pycharm cheatsheet](https://resources.jetbrains.com/storage/products/pycharm/docs/PyCharm_ReferenceCard.pdf) that can help you with a lot of extra shortcuts. Remember all shortcuts can be applied to all Jetbrains IDEs.
+
+## Visual Studio Code shortcuts
+
+Visual Studio Code provides a rich and easy keyboard shortcuts editing experience using Keyboard Shortcuts editor. It lists all available commands with and without keybindings and you can easily change / remove / reset their keybindings using the available actions. It also has a search box on the top that helps you in finding commands or keybindings. You can open this editor by going to the menu under **File** > **Preferences** > **Keyboard Shortcuts**. (**Code** > **Preferences** > **Keyboard Shortcuts** on macOS).
+
+Most importantly, you can see keybindings according to your keyboard layout. For example, key binding `Cmd+\` in US keyboard layout will be shown as `Ctrl+Shift+Alt+Cmd+7` when layout is changed to German. The dialog to enter key binding will assign the correct and desired key binding as per your keyboard layout.
+
+### Fold and Unfold
+
+| Function | Shortcut |
+| ------------------------------- | -------------------------------- |
+| Fold (collapse) region | `Ctrl+Shift+[` |
+| Unfold (uncollapse) region | `Ctrl+Shift+]` |
+| Fold (collapse) all subregions | `Ctrl+K Ctrl+[` |
+| Unfold (uncollapse) all subregions | `Ctrl+K Ctrl+]`|
+| Fold (collapse) all regions | `Ctrl+K Ctrl+0` |
+| Unfold (uncollapse) all regions | `Ctrl+K Ctrl+J` |
+
+### Rich Languages Editing
+
+| Function | Shortcut |
+| ------------------------------- | -------------------------------- |
+| Trigger Suggest | `Ctrl+Space` |
```

Comment:
Does not work in MacOSx if you have multiple languages

Reply:
Thanks for the suggestion Wlady, @jase156 can you help with this, please?

Reply:
@Wason1797 I put only the windows/linux commands in this section as a summary, but at the end there is a reference to the different documentation for each OS. If we put for mac, linux and windows it would be very long, or do you want to put it?